### PR TITLE
Make config (mostly) non-global

### PR DESCRIPTION
**Pull request by A1Liu**
- Make project config non-global, except for the package JSON thing with project name
- Add tests for App config loading
- Add `LoadFromEnv` to load a project from the global project path

### CODE REVIEW

File: backend/internal/project/app_config_test.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import (
+	"net/url"
+	"path/filepath"
+	"testing"
+)
+
+func TestAppConfigLoad(t *testing.T) {
+	projectPath := t.TempDir()
+	err := createProjectStructure(projectPath, map[string]string{
+		// TODO: Weird things happen when passing in `file:app2/robin.app.json`
```
Comment (karimsa) [inline backend/internal/project/app_config_test.go:12]: Did you try `file://...`? `file:...` is not a valid URL

**Review by karimsa — CHANGES_REQUESTED** [review]
This makes the API for loading the project path and/or config unnecessarily verbose - I get why you want the lower level API for tests, but there should be a global for actual runtime usage.

**Comment by A1Liu** [comment]
Yee okie ill add global

File: backend/internal/project/app_config_test.go (hunk 1 above)
Comment (A1Liu) [reply backend/internal/project/app_config_test.go:12]: I tried both, the thing is that in both cases, the URL parse succeeds but the opening fails.

**Review by karimsa — APPROVED** [review]

**Pull request merged**